src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -574,35 +691,104 @@ private void AddSchema(CsdlSchema schema, bool addAnnotations)
             {
                 RegisterElement(container);
             }
+        }
 
-            if (!string.IsNullOrEmpty(schema.Alias))
+        /// <summary>
+        /// Adds out-of-line annotations from the specified schema
+        /// if they are referenced in the <paramref name="includeAnnotationsIndex"/> cache.
+        /// </summary>
+        /// <param name="schema">The schema to add annotations from.</param>
+        /// <param name="schemaWrapper">The <see cref="CsdlSemanticsSchema"/> wrapper for the provided schema</param>
+        /// <param name="includeAnnotationsIndex">Cache of the schema's out-of-line <see cref="IEdmIncludeAnnotations"/>s indexed by the term namespace.
+        /// If this dictionary is non-empty, then only annotations that match
+        /// the references will be added. If it's empty, no annotations will be added. If it's null, all annotations
+        /// will be added unconditionally.</param>
+        private void AddOutOfLineAnnotationsFromSchema(
+            CsdlSchema schema,
+            CsdlSemanticsSchema schemaWrapper,
+            Dictionary<string, List<IEdmIncludeAnnotations>> includeAnnotationsIndex)
+        {
+            if (includeAnnotationsIndex?.Count == 0)

```

Comment:
```suggestion
            if (includeAnnotationsIndex?.Any() == false)

```

Reply:
or handle the null in a better way as Any can cause some issues with performance here due. I believe `null == 0` wil give you false but I could be wrong.

Reply:
I didn't understand this comment very well. But I made the null check more explicit `includeAnnotationsIndex != null && includeAnnotationsIndex.Count == 0`